Lacework/lacework_module/lacework_connector.py

```diff
@@ -121,55 +138,58 @@ def get_next_events(self, cursor: str | None) -> Any | None:
-        cursor = self.cursor
-
-        messages = []
-        while has_more_messages and self.running:
-            has_more_messages = False
-
-            batch = self.get_next_events(cursor)
-            if batch is None:
-                break
+        now = datetime.now(timezone.utc)
 
-            if batch.get("content", {}).get("paging", {}).get("urls", {}).get("nextPage") != None:
-                has_more_messages = True
-                cursor = max(item.get("startTime") for item in batch.get("data", []))
+        first_batch = self.get_response_by_timestamp()
+        if first_batch:
+            next_page_url = first_batch.get("content", {}).get("paging", {}).get("urls", {}).get("nextPage")
+            items = first_batch.get("data", [])
 
-            items = batch.get("data", [])
-            if len(items) > 0:
-                most_recent_timestamp_seen = self._get_most_recent_timestamp_from_items(items)
-                events_lag = int(time.time() - most_recent_timestamp_seen)
-                EVENTS_LAG.labels(intake_key=self.configuration.intake_key).set(events_lag)
-                INCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(items))
+            last_date = self._get_most_recent_timestamp_from_items(items)
+            events_lag = int(time.time() - last_date)
+            EVENTS_LAG.labels(intake_key=self.configuration.intake_key).set(events_lag)
 
-            for message in items:
-                messages.append(orjson.dumps(message).decode("utf-8"))
-
-            if len(messages) > self.pagination_limit:
-                self.log(message=f"Sending a batch of {len(messages)} messages", level="info")
-                OUTCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(messages))
-                self.push_events_to_intakes(events=messages)
-                messages = []
-
-        self.cursor = cursor
-
-        if messages:
-            self.log(message=f"Sending a batch of {len(messages)} messages", level="info")
+            messages = [orjson.dumps(message).decode("utf-8") for message in items]
+            self.log(message=f"Sending the first batch of {len(messages)} messages", level="info")
             OUTCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(messages))
             self.push_events_to_intakes(events=messages)
+
+            grouped_data = []
+            while next_page_url:
+                response_next_page = self.get_next_events(next_page_url)
+                if response_next_page:
+                    next_page_url = (
+                        response_next_page.get("content", {}).get("paging", {}).get("urls", {}).get("nextPage")
+                    )
+                    next_page_items = response_next_page.get("items", [])
+                    next_page_items = [orjson.dumps(message).decode("utf-8") for message in next_page_items]
+                    grouped_data.extend(next_page_items)
```

Comment:
We have to compute the lag for each batch
```suggestion
                    grouped_data.extend(next_page_items)
                     last_date = self._get_most_recent_timestamp_from_items(next_page_items)
                    events_lag = int(time.time() - last_date)
                    EVENTS_LAG.labels(intake_key=self.configuration.intake_key).set(events_lag)
                    self.most_recent_date_seen = last_date
```